awx/main/analytics/core.py

```diff
@@ -26,6 +26,14 @@
 manifest = dict()
 
 
+def _valid_config():
+    for key in ['AUTOMATION_ANALYTICS_URL', 'REDHAT_USERNAME', 'REDHAT_PASSWORD']:
+        if not getattr(settings, key, None):
+            logger.error('%s is not set' % (key,))
+            return False
+    return True
```

Comment:
@wenottingham this covers case where username/password not set but does not cover case where they are invalid. Do we want to try that out?

Also, if I'm reading this correctly we would still allow the user to enabled automation analytics w/o username/password or invalid username/password as we do today but now we would log this error when we then are unable to proceed because of missing auth.

I'm guessing we would want to make a backlog item for making this more visible in the UI if automation analytics is enabled but failing because of this?

Reply:
Testing validity would be making a random connection that didn't send anything, I'm not sure yet if i want to cause logs/errors on the insights data gathering side for just testing a connection.

Correct, we may want to surface this higher in the future; this is just to stop it from even trying and wasting resources if they didn't configure it right.